modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/data_set_display/DataSetDisplay.js

```diff
@@ -394,7 +394,7 @@ function DataSetDisplay({
 		return executeAsyncAction(url, method)
 			.then((_) => {
 				if (isMounted()) {
-					refreshData();
+					delay(500).then(refreshData);
```

Comment:
Hey @FabioDiegoMastrorilli, I think we should move the isMounted() check to inside the delay callback to be on the safe sie:

```js
return executeAsyncAction(url, method)
    .then((_) => {
        delay(500).then(() => {
            if (isMounted()) {
                refreshData();
            } 
        });

        ...
    })
```

Reply:
You're totally right

Reply:
Instead of the `delay` that you are using:

```js
function delay(duration) {
	return new Promise((resolve) => {
		setTimeout(() => resolve(), duration);
	});
}
```

You could also use `useTimeout` from `frontend-js-web`; it has an `isMounted()` check bulit into it.

```js
import {useTimeout} from 'frontend-js-web';

// then...

const delay = useTimeout();

// and...

delay(refreshData, 500);
```